### PR TITLE
BKNDLSS-29018 Simple Modal

### PR DESCRIPTION
**Pull request by StasivPavlo**
input_dark-fire
input
input_active
alert
confirm
input_dark

### CODE REVIEW

File: components/bl-simpleModal-component/component.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
+{
+  "id": "c_25c90a88da567672578c1b7c8d4e44a9",
+  "name": "Simple Modal",
+  "description": "",
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:4]: add description, please check PR checklist

File: components/bl-simpleModal-component/component.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  "category": "Custom Components",
+  "properties": [
+    {
+      "type": "text",
+      "name": "titleModal",
+      "label": "Title Modal",
+      "handlerId": "titleModalLogic",
+      "handlerLabel": "Title Modal Logic",
+      "dataBinding": true,
+      "handlerDescription": "This is a handler for the logic to determine the title for SImple Modal."
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:18]: This is a handler for the logic to determine the title.

File: components/bl-simpleModal-component/component.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    {
+      "type": "text",
+      "name": "textModal",
+      "label": "Text Modal",
+      "handlerId": "textModalLogic",
+      "handlerLabel": "Text Modal Logic",
+      "dataBinding": true,
+      "handlerDescription": "This is a handler for the logic to determine the text for SImple Modal."
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:27]: text -> content

File: components/bl-simpleModal-component/src/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  if (!display || !isModalOpen) {
+    return null;
+  }
+
+  return (
+    <div className={ 'bl-customComponent-simple-modal ' + classList.join(' ') }>
+      <div className={ simpleModalClasses } style={ { animationDuration: `${ closingDuration }s` } }>
+        <div onClick={ onCloseButton } className="overlay"></div>
+        <div className="simple-modal__content">
+          { titleModal && (
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:59]: In external component?

File: components/bl-simpleModal-component/src/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+              inputValue={ inputValue }
+              setInputValue={ setInputValue }
+              placeholderText={ placeholderText }
+            />
+          ) }
+
+          <SimpleModalButton
+            typeSimpleModal={ typeSimpleModal }
+            onCloseButton={ onCloseButton }
+            onSubmitButton={ onSubmitButton }
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:77]:
```
onClose={ onClose }
onSubmit={ onSubmit }
```

File: components/bl-simpleModal-component/src/index.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  component.openModal = () => {
+    setIsModalOpen(true);
+  };
+
+  useEffect(() => {
+    onInputValueChange({ inputValue });
+  }, [inputValue]);
+
+  if (isModalOpen) {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:44]: into `useEffect` with `isModalOpen` deps

File: components/bl-simpleModal-component/src/index.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  } = component;
+  const { onCloseButton, onSubmitButton, onInputValueChange } = eventHandlers;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const simpleModalClasses = useSimpleModalClasses(isClosing);
+
+  component.closeModal = () => {
+    setIsClosing(false);
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:24]: When we close the modal window that must be in 'closing' state, so for me, it's not logical when we close the modal window but set isClosing to false.

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-simpleModal-component/component.json [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      "name": "titleModal",
+      "label": "Title Modal",
+      "handlerId": "titleModalLogic",
+      "handlerLabel": "Title Modal Logic",
+      "dataBinding": true,
+      "handlerDescription": "This is a handler for the logic to determine the title."
+    },
+    {
+      "type": "text",
+      "name": "textModal",
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:22]: "name": "content",
"label": "Contentl",
"handlerId": "contentLogic",
"handlerLabel": "Content Logic",

File: components/bl-simpleModal-component/component.json [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        },
+        {
+          "value": "confirm",
+          "label": "Confirm"
+        }
+      ]
+    },
+    {
+      "type": "text",
+      "name": "placeholderText",
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:61]: placeholderText -> placeholder

File: components/bl-simpleModal-component/component.json [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+      "label": "Placeholder Text",
+      "handlerId": "placeholderTextLogic",
+      "handlerLabel": "Placeholder Text Logic",
+      "dataBinding": true,
+      "handlerDescription": "This is a handler for the logic to determine the text for input placeholder."
+    },
+    {
+      "type": "text",
+      "name": "submitButtonLabel",
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/component.json:70]: all labels should be near each other

File: components/bl-simpleModal-component/src/index.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react';
+import { SimpleModalButton, SimpleModalContainer, SimpleModalTitle } from './subcomponents';
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:2]: Button, Container, Title

File: components/bl-simpleModal-component/styles/index.less [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+
+    width: 100vw;
+    height: 100vh;
+
+    background: rgba(0, 0, 0, 0.5);
+
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/styles/index.less:32]: remove empty line

File: components/bl-simpleModal-component/README.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+<dl>
+<dt>Title</dt>
+<dd>Allows write title for Simple Modal.</dd>
+<dt>Content</dt>
+<dd>Allows write content for Simple Modal</dd>
+<dt>Close Button Label</dt>
+<dd>Allows to write label for close button. Default value "Close"</dd>
+<dt>Type</dt>
+<dd>Allows select type of Simple Modal (Alert, Prompt, Confirm). Default value 'Alert'.</dd>
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/README.md:18]: Allows select type of Simple Modal ("alert", "prompt", "confirm"). Default value "alert".

File: components/bl-simpleModal-component/src/index.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+  if (!display || !isModalOpen) {
+    return null;
+  }
+
+  return (
+    <div className={ rootClasses }>
+      <div className={ modalClasses } style={ { animationDuration: `${ closingDuration }ms` } }>
+        <div onClick={ onClose } className="overlay"></div>
+        <div className="simple-modal__content">
+          { title && (<Title title={ title }/>) }
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:59]: just `<Title title={ title }/>`

and in component
```
if(!title) {
 return null
}
```

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+        >
+          { placeholder }
+        </label>
+      </div>
+    </div>
+  );
+}
+
+export function Container(props) {
+  const {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:36]: in one line

File: components/bl-simpleModal-component/src/index.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+    setIsClosing(false);
+  };
+
+  useEffect(() => {
+    onInputValueChange({ inputValue });
+  }, [inputValue]);
+
+  useEffect(() => {
+    if (isModalOpen) {
+      document.body.classList.toggle('active-modal');
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/index.js:44]: Please read how classList.toggle works

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 17]
```diff
@@ -0,0 +1,112 @@
+function SimpleModalInput(props) {
+  const {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:2]: in one line

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+      { content && (
+        <p className="simple-modal__text">
+          { content }
+        </p>
+      ) }
+
+      { type === 'prompt' && (
+        <SimpleModalInput
+          inputValue={ inputValue }
+          onInputValue={ setInputValue }
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:45]: onInputValue onInputChange onChange onInputValueChange - could we make all this handler names consistent?

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+
+  return (
+    <h2 className="simple-modal__title">
+      { title }
+    </h2>
+  );
+}
+
+export function Button(props) {
+  const {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:68]: in one line

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+  if (title) {
+    return null;
+  }
+
+  return (
+    <h2 className="simple-modal__title">
+      { title }
+    </h2>
+  );
+}
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:65]:
```
export function Title({ content }) {
  if (!content) {
    return null;
  }

  return (
    <h2 className="simple-modal__title"> { content } </h2>
  );
}
```

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+        />
+      ) }
+    </div>
+  );
+}
+
+export function Title(props) {
+  const { title } = props;
+
+  if (title) {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:56]: may be `!title`?

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+  const {
+    type,
+    onClose,
+    onSubmit,
+    inputValue,
+    submitButtonLabel,
+    closeButtonLabel,
+  } = props;
+
+  const handlerSubmitted = () => {
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:77]: onSubmitHandler

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 23]
```diff
@@ -0,0 +1,106 @@
...
+    <div className="simple-modal__button-container">
+      <button
+        type="button"
+        className="simple-modal__button"
+        onClick={ onClose }
+      >
+        { closeButtonLabel }
+      </button>
+
+      { (type === 'prompt' || type === 'confirm') && (
```
Comment (v-excelsior) [inline components/bl-simpleModal-component/src/subcomponents.js:95]: Modal types should be constant

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-simpleModal-component/src/index.js [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+    type,
+    placeholder,
+    closeButtonLabel,
+    submitButtonLabel,
+    closingDuration,
+  } = component;
+  const { onClose, onSubmit, onInputValueChange } = eventHandlers;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isClosing, setIsClosing] = useState(true);
```
[the remark this answers is not in the document]
Comment (StasivPavlo) [reply components/bl-simpleModal-component/src/index.js:19]: component by default is closed. isClosing = true it means that the component is closed. isClosing needed for activating closing animation. Only "display" and "isModalOpen" are responsible for the visibility of the component.

File: components/bl-simpleModal-component/src/index.js [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+    setIsModalOpen(true);
+    setIsClosing(false);
+  };
+
+  useEffect(() => {
+    onInputValueChange({ inputValue });
+  }, [inputValue]);
+
+  useEffect(() => {
+    document.body.classList.toggle('active-modal', isModalOpen);
```
[the remark this answers is not in the document]
Comment (StasivPavlo) [reply components/bl-simpleModal-component/src/index.js:43]: if the actions of both components are activated together

File: components/bl-simpleModal-component/src/index.js (hunk 24 above)
[the remark this answers is not in the document]
Comment (v-excelsior) [reply components/bl-simpleModal-component/src/index.js:20]: >isClosing = true it means that the component is closed.

The reason I ask about it "is closing" - does not mean "it closed now", it mean's "it in the closing process now", that grammatically incorrect.
image

File: components/bl-simpleModal-component/src/index.js [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+  component.openModal = () => {
+    setIsOpen(true);
+  };
+
+  useEffect(() => {
+    onInputValueChange({ inputValue });
+  }, [inputValue]);
+
+  useEffect(() => {
+    document.body.classList.toggle('active-modal', isOpen);
```
Comment (Valodya) [inline components/bl-simpleModal-component/src/index.js:44]: should not this component use https://reactjs.org/docs/portals.html ?

File: components/bl-simpleModal-component/src/subcomponents.js [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+  if (!content) {
+    return null;
+  }
+
+  return (
+    <h2 className="simple-modal__title">{ content }</h2>
+  );
+}
+
+export function Button(props) {
```
Comment (Valodya) [inline components/bl-simpleModal-component/src/subcomponents.js:65]: Button => Buttons/Controls/ModalButtons/ModalControls?

File: components/bl-simpleModal-component/styles/index.less [hunk 28]
```diff
@@ -0,0 +1,208 @@
...
+@bl-customComponent-simple-modal-button-backgroundColor: transparent;
+@bl-customComponent-simple-modal-button-backgroundColorHover: fade(@bl-customComponent-simple-modal-themeColor, 4%);
+@bl-customComponent-simple-modal-button-borderRadius: 4px;
+
+body.active-modal {
+  overflow-y: hidden;
+}
+
+.bl-customComponent-simple-modal {
+  z-index: 1;
```
Comment (Valodya) [inline components/bl-simpleModal-component/styles/index.less:19]: I assume some components might have higher z-index and as a result they can overlay the modal 

however, with using ReactProtal the problem should not appear

File: components/bl-simpleModal-component/src/modal.js [hunk 29]
```diff
@@ -0,0 +1,82 @@
...
+  const root = useMemo(() => {
+    return document.createElement('div');
+  }, [])
+  root.classList.add(cn('bl-customComponent-simple-modal', classList));
+
+  useEffect(() => {
+    document.body.appendChild(root);
+  }, []);
+
+  useEffect(() => {
```
Comment (Valodya) [inline components/bl-simpleModal-component/src/modal.js:32]:
```
useEffect(() => {
    document.body.appendChild(root);

    return () => {
      document.body.removeChild(root);
    };
  });
```

**Pull request merged**